File: Лабораторні/src/lab-02/datetime.pas
program DateTimeCGI;

{$mode objfpc}{$H+}

uses
  SysUtils,
  Classes,
  DateUtils;

begin
  // Виведення заголовка Content-Type
  writeln('Content-Type: text/html');
  writeln;

  // Отримання поточної дати та часу
  let CurrentDateTime: TDateTime := Now;

  // Форматування дати та часу
  let DateTimeStr: string := FormatDateTime('dd.mm.yyyy hh:nn:ss', CurrentDateTime);

  // Виведення HTML-сторінки з поточною датою та часом
  writeln('<html><head><title>Дата та час</title></head><body>');
  writeln('<h1>Поточна дата та час:</h1>');
  writeln('<p>', DateTimeStr, '</p>');
  writeln('</body></html>');
end.

File: Лабораторні/src/lab-02/datetime.cpp
#include <iostream>
#include <ctime>

int main() {
    // Виведення заголовка Content-Type
    std::cout << "Content-Type: text/html\r\n\r\n";

    // Отримання поточного часу
    time_t currentTime;
    struct tm* timeInfo;
    time(&currentTime);
    timeInfo = localtime(&currentTime);

    // Виведення HTML-сторінки з поточною датою та часом
    std::cout << "<html><head><title>Дата та час</title></head><body>";
    std::cout << "<h1>Поточна дата та час:</h1>";
    std::cout << "<p>" << asctime(timeInfo) << "</p>";
    std::cout << "</body></html>";

    return 0;
}
